packages/flutter_tools/lib/src/test/flutter_platform.dart

```diff
@@ -105,8 +112,16 @@ void main() {
 }
 ''');
 
+    int observatoryPort;
```

Comment:
My understanding was the dart vm would pick an open port if you passed --observatory-port=0, and then we had code to read said port from the logs, etc.  Maybe that won't work through our package:test integration?  Devon might know.